### PR TITLE
Move what-is-kubernetes to overview page

### PR DESCRIPTION
**Pull request by Rishit-dagli**
Move https://kubernetes.io/docs/concepts/overview/what-is-kubernetes/ to https://kubernetes.io/docs/concepts/overview/

This PR also makes sure that:

- a new redirect
- make sure that the table of contents is no longer rendered

Fixes: #35251

### CODE REVIEW

**Comment by tengqm** [comment]
/label refactor

Changes like this need a discussion across the team.

File: content/en/docs/concepts/overview/_index.md [hunk 1]
```diff
@@ -1,7 +1,97 @@
...
+description: >
+  Kubernetes is a portable, extensible, open source platform for managing containerized workloads and services, that facilitates both declarative configuration and automation. It has a large, rapidly growing ecosystem. Kubernetes services, support, and tools are widely available.
+content_type: concept
 weight: 20
-description: Get a high-level outline of Kubernetes and the components it is built from.
+card:
+  name: concepts
+  weight: 10
 sitemap:
   priority: 0.9
```
Comment (sftim) [inline content/en/docs/concepts/overview/_index.md:14]: After the move, should this 0.9 priority be updated?

File: content/en/docs/concepts/overview/components.md [hunk 2]
```diff
@@ -4,8 +4,8 @@ reviewers:
 title: Kubernetes Components
 content_type: concept
 description: >
-  A Kubernetes cluster consists of the components that represent the control plane
-  and a set of machines called nodes.
+  A Kubernetes cluster consists of the components that are a part of the control
+  plane and a set of machines called nodes.
```
Comment (sftim) [inline content/en/docs/concepts/overview/components.md:8]: This change makes sense to me.

File: content/en/docs/concepts/overview/_index.md [hunk 3]
```diff
@@ -1,7 +1,97 @@
...
+* Does not provide nor mandate a configuration language/system (for example, Jsonnet). It provides a declarative API that may be targeted by arbitrary forms of declarative specifications.
+* Does not provide nor adopt any comprehensive machine configuration, maintenance, management, or self-healing systems.
+* Additionally, Kubernetes is not a mere orchestration system. In fact, it eliminates the need for orchestration. The technical definition of orchestration is execution of a defined workflow: first do A, then B, then C. In contrast, Kubernetes comprises a set of independent, composable control processes that continuously drive the current state towards the provided desired state. It shouldn't matter how you get from A to C. Centralized control is also not required. This results in a system that is easier to use and more powerful, robust, resilient, and extensible.
+
+
+
+## {{% heading "whatsnext" %}}
+
+*   Take a look at the [Kubernetes Components](/docs/concepts/overview/components/)
+*   Ready to [Get Started](/docs/setup/)?
```
Comment (sftim) [inline content/en/docs/concepts/overview/_index.md:97]: IMO we now also need links to:
- `/docs/concepts/overview/kubernetes-api/`
- `/docs/concepts/architecture/`

**Review by sftim — COMMENTED** [review]
I'm happy with the intent here.
https://deploy-preview-35298--kubernetes-io-main-staging.netlify.app/docs/concepts/overview/ looks much better than https://kubernetes.io/docs/concepts/overview/ (to me)

File: content/en/docs/concepts/overview/_index.md (hunk 3 above)
Comment (Rishit-dagli) [reply content/en/docs/concepts/overview/_index.md]: gotcha, fixed this

File: content/en/docs/concepts/overview/_index.md (hunk 1 above)
Comment (Rishit-dagli) [reply content/en/docs/concepts/overview/_index.md:14]: Ah, yes, fixed this as well

**Review by mehabhalodiya — APPROVED** [review]
Looks like all pointers mentioned in https://github.com/kubernetes/website/issues/35251#issuecomment-1192783833 has been addressed.

/lgtm

**Comment by Rishit-dagli** [comment]
/assign @kcmartin
> Once this PR has been reviewed and has the lgtm label, please assign kcmartin for approval

**Review by sftim — COMMENTED** [review]
Thanks

/approve

**Pull request merged**